appendices.tex

```diff
@@ -202,41 +203,105 @@ \subsubsection{Migrating your ledger}
 \section{Migrating between PostgreSQL versions}
 \label{sec-migration-postrgesql}
 
-If you're migrating between PostgreSQL versions, there are a few things to take
-into account.
+There are generally 2 ways to upgrade PostgreSQL\index{PostgreSQL upgrade} that apply to normal LSMB installations. 
+Both are described in the current PostgreSQL documentation at \url{https://www.postgresql.org/docs/current/index.html}.
+\begin{enumerate}
+    \item Dump\index{dump} and restore\index{restore} -
+        Described in more detail below.
+        Note that the roles must be restored before database creation and the data restore.
+    \item Upgrade in place using \texttt{pg\_upgrade}\index{pg\_upgrade} - When doing an upgrade in place, there are no specific LedgerSMB requirements.
+        Beware, upgrade in place requires both versions of the PostgreSQL to be installed and working. It is therefore more complex.
+\end{enumerate}
+
+LedgerSMB versions prior to version 1.9.16 did not support PostgreSQL 14 or higher due to a backward incompatible change in PostgreSQL 14.
 
-% pg_dump: use the NEW pg_dump utility against the old database
-% pg_upgrade: <TODO>
-% pg_<migratecluster>: <TODO>; in case of unclean extension upgrade,
-%              possible need to issue ``CREATE EXTENSION tablefunc FROM UNPACKAGED''
+The following examples use the  \texttt{lsmb\_dbadmin} for the database administrative user. 
+Recent LedgerSMB installs do not automatically create an  \texttt{lsmb\_dbadmin}\index{lsmb\_dbadmin} user.

```

Comment:
Older ones didn't either... Creating a lsmb_admin user is our advice because it prevents using the super user, meaning that even though this user *can* create new users, we are 100% sure it won't ever be a super user.

The lsmb_admin user isn't a precondition, though. We consider it general good security practice.

Reply:
Done

Reply:
> What is the procedure and permissions required for lsmb_admin to do backup and restore? Is this documented someplace?

It not being a super user,  you should be able to back up a database that's created with lsmb_admin, however, a database created by any other user won't be possible to back up with lsmb_admin, unless it's a super user.

It's flat out impossible to create a role backup with lsmb_admin, because roles are so-called "globals" and you need a super user to access *all* globals and thus a backup of them.